org.eclipse.buildship.core/src/main/java/org/eclipse/buildship/core/preferences/internal/DefaultModelPersistence.java

```diff
@@ -54,32 +52,33 @@
 public final class DefaultModelPersistence implements ModelPersistence, EventListener {
 
     // Cache holding (project name - persistent model property) pairs.
```

Comment:
I guess that would work too. If we are inside our synchronization code, we will definitely update the state anyway. The difference between present and default values is only relevant to user code.